src/sqlfluff/rules/L005.py

```diff
@@ -41,20 +41,22 @@ class Rule_L005(BaseRule):
     """
 
     groups = ("all", "core")
-    crawl_behaviour = SegmentSeekerCrawler({"comma"}, provide_raw_stack=True)
+    crawl_behaviour = SegmentSeekerCrawler({"comma"})
 
     def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Commas should not have whitespace directly before them."""
-        if not context.raw_stack:
-            return None  # pragma: no cover
-        anchor: Optional[RawSegment] = context.raw_stack[-1]
-        if (
-            # We need at least one segment previous segment for this to work.
-            anchor is not None
-            and context.segment.is_type("comma")
-            and anchor.is_type("whitespace")
-            and anchor.pos_marker.line_pos > 1
-        ):
-            return LintResult(anchor=anchor, fixes=[LintFix.delete(anchor)])
-        # Otherwise fine.
+        fixes = (
+            ReflowSequence.from_around_target(
+                context.segment,
+                context.parent_stack[0],
```

Comment:
Nitpick: I notice the rules always pass the same 2nd and 3rd parameters:
```
context.parent_stack[0],
config=context.config,
```

Did you consider letting the rules just pass `context` and having it get `.parent_stack[0]` and `.config` from that?

Reply:
I did, but I'd rather not pass around the whole context - it's quite powerful and it makes it a little harder to separate concerns, because you don't know which bits of the context they're using.

It was quite hard to unpick some of the older implementations for exactly that reason!